lib/rmg/WaterProxy.cpp

```diff
@@ -142,6 +142,13 @@ RouteInfo WaterProxy::waterRoute(Zone & dst)
 				dst.areaPossible().subtract(lake.neighbourZones[dst.getId()]);
 				continue;
 			}
+
+			//Don't place shipyard or boats on the very small lake
+			if (lake.area.getTiles().size() < 25)
```

Comment:
Are we sure this won't lead to unplayable maps?

Reply:
In an edge case it should fall back to Monolith conection, but I'll double check.

Reply:
Yep, the default route is invalid - so if RMG won't find any other connection, it will just place Monolith pair.